src/eduid_userdb/reset_password/db.py

```diff
@@ -0,0 +1,170 @@
+from eduid_userdb.reset_password import ResetPasswordEmailState, ResetPasswordEmailAndPhoneState
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+class ResetPasswordUserDB(UserDB):
+
+    UserClass = ResetPasswordUser
+
+    def __init__(self, db_uri: str,
+                 db_name: str = 'eduid_reset_password',
+                 collection: str = 'profiles'):
+        super(ResetPasswordUserDB, self).__init__(db_uri, db_name,
+                                                  collection=collection)
+
+    def save(self, user: User, check_sync: bool = True,
+             old_format: bool = False) -> bool:
+        return super(ResetPasswordUserDB, self).save(user, check_sync=check_sync,
+                                                     old_format=old_format)
+
+
+class ResetPasswordStateDB(BaseDB):
+
+    def __init__(self, db_uri: str,
+                 db_name: str = 'eduid_reset_password',
+                 collection: str = 'password_reset_data'):
+        super(ResetPasswordStateDB, self).__init__(db_uri, db_name,
+                                                   collection=collection)
+
+    def get_state_by_email_code(self, email_code: str,
+                                raise_on_missing: bool = True):
+        """
+        Locate a state in the db given the state's email code.
+
+        :param email_code: Code sent to the user
+        :param raise_on_missing: Raise exception if True else return None
+
+        :return: ResetPasswordState instance | None
+
+        :raise self.DocumentDoesNotExist: No document match the search criteria
+        :raise self.MultipleDocumentsReturned: More than one document matches
+                                               the search criteria
+        """
+        spec = {'email_code.code': email_code}
+        states = list(self._get_documents_by_filter(spec,
+                                            raise_on_missing=raise_on_missing))
+
+        if len(states) == 0:
+            return None
+
+        if len(states) > 1:
+            raise MultipleDocumentsReturned(f"Multiple matching users for "
+                                            f"filter {filter}")
+
+        return self.init_state(states[0])
+
+    def get_state_by_eppn(self, eppn: str, raise_on_missing: bool = True):
+        """
+        Locate a state in the db given the users eppn.
+
+        :param eppn: Users unique eppn
+        :param raise_on_missing: Raise exception if True else return None
+
+        :return: ResetPasswordState instance | None
+
+        :raise self.DocumentDoesNotExist: No document match the search criteria
+        :raise self.MultipleDocumentsReturned: More than one document matches
+                                               the search criteria
+        """
+        state = self._get_document_by_attr('eduPersonPrincipalName', eppn,
+                                           raise_on_missing)
+        if state:
+            return self.init_state(state)
+
+    @staticmethod
```

Comment:
Would it not be better with at classmethod?

Reply:
Not sure what you mean - To use the provided cls to instantiate the `resetpw_state`? Then we'd be overriding whatever comes in `state.get('method')`, is that right?